File: data_generate/build_schools.js
const { Pool, Client } = require('pg')
const connectionString = 'postgresql://localhost/Cathal'


const pool = new Pool({
    connectionString: connectionString,
    max: 10000
})

function clean(text) {
    var cleaned = null;
    try {
        cleaned = text.replace(/'/g, "''");
    }catch(e){
        return null;
    }

    return cleaned;
}


pool.query(`SELECT DISTINCT school FROM module`, (err, res) => {
    const rows = res.rows;

    console.log(rows)

    //var row1 = rows[1100];

    //console.log(row1);
    for (var i = 0; i < rows.length; i++) {
        var row = rows[i];
        var count_modules = 0;
        var count_staff = 0;
        var count_subjects = 0;
        console.log(row.school)
      
        pool.query('SELECT code FROM module'), (err, res) => {
            console.log(err)
            console.log(res)
            // count_modules = res.rows[0].count;
            // console.log(count_modules)
        }
    }

})

File: data_generate/build_schools.py
import psycopg2

psycopg2.extensions.register_type(psycopg2.extensions.UNICODE)
psycopg2.extensions.register_type(psycopg2.extensions.UNICODEARRAY)


conn = psycopg2.connect("dbname=Cathal host=localhost port=5432")


cur = conn.cursor()
cur.execute("SELECT DISTINCT school FROM module")
schools = cur.fetchall()
    
print schools

for school in schools:
    school_name = school[0].encode('utf8')
    url_name = ''.join(e for e in school_name if e.isalnum())
    print school_name
    print url_name
    cur.execute("""SELECT COUNT(code) FROM module WHERE school = '%s'""" % school_name)
    count_modules = cur.fetchone()[0]
    print 'Modules:', count_modules

    cur.execute("""SELECT COUNT(DISTINCT module_coordinator) FROM module WHERE school = '%s'""" % school_name)
    count_staff = cur.fetchone()[0]
    print 'Staff:', count_staff

    cur.execute("""SELECT COUNT(DISTINCT subject) FROM module WHERE school = '%s'""" % school_name)
    count_subjects = cur.fetchone()[0]
    print 'Subjects:', count_subjects

    cur.execute("""INSERT INTO school (name, url_name, count_subjects, count_staff, count_modules) VALUES ('%s','%s',%s,%s,%s)""" % (school_name, url_name, count_subjects, count_staff, count_modules))

conn.commit()
conn.close()